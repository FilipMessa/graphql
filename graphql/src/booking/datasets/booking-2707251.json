{
  "original_price": 74.61,
  "grade": {
    "id": "A",
    "time": 30
  },
  "additional_contacts": [],
  "cancel_requested": null,
  "blocked": false,
  "additional_bags_prices": {
    "1": 23.65,
    "2": 47.3
  },
  "segments": null,
  "flights": [
    {
      "arrival": {
        "when": {
          "utc": 1504992600,
          "local": 1504999800
        },
        "where": {
          "city_id": "paris_fr",
          "country": "FR",
          "code": "CDG",
          "name": "Paris"
        }
      },
      "bags_recheck_required": false,
      "return": 0,
      "flight_no": 8773,
      "ticket_available_at": "2017-09-02",
      "reservation_number": "chat",
      "id": "315289498",
      "guarantee": false,
      "departure": {
        "when": {
          "utc": 1504987800,
          "local": 1504991400
        },
        "where": {
          "city_id": "london_gb",
          "country": "GB",
          "code": "LGW",
          "name": "London"
        }
      },
      "airline": {
        "checkin": 7,
        "iata": "VY",
        "doing_online_checkin": 1,
        "name": "Vueling"
      },
      "operating_airline": {
        "iata": "VY",
        "name": "Vueling"
      }
    }
  ],
  "grace_period_fee": null,
  "original_flight_price": null,
  "flight_ids": [
    "315289498"
  ],
  "original_currency": "USD",
  "bid": 2707251,
  "bags_restrictions": {
    "holds": {},
    "hands": {}
  },
  "podio_status": 32,
  "card_payment": 69,
  "prices": [
    "30.0"
  ],
  "pending_additional_services": [],
  "transfer_ids": [],
  "booking_token": "GamFTRHoBQ/v6QB8IrykhuGglDtGWxUFRVNsUQAg/ZpAsAG2Oqe2XPJISBND8WVXgo5DZRAIYNiKixd3TRfLXLHGOBVDNZ85k4qWBOxjqWc1fM/4eZLzZov2mfgi6+6LCjNG2T8YnXsIdPWzWxQWV3Rao3y+8VsgO3WJxNKzNliu5rU9sbO7tjyA3K0iDC4QBC/k6Xg59t6elZs8l29NR/NODhA+jFr6ZPjWzDWHRpuMnEeWEsSNY7Ffpw352VLeROBltX4IzHLPly15sLoJbnX1g7B1MGT/TKNudpPT0jsCzcMQ8pz2P8efRYuvderzhbE9O4CT6xxba2GLZ8v5Zie4mOOSB6nrRfFinHBjz7M=",
  "checkin_deadline": 1504728000,
  "pending_refunds": [],
  "bag_params": {
    "hand2_note": "",
    "hand_length": 55,
    "hold_height": 79,
    "hand2_width": null,
    "hold_weight": 23,
    "hand_note": "",
    "hold_length": 250,
    "hand2_length": null,
    "note": "max 180cm (length + width + height)",
    "hand_height": 20,
    "hand2_weight": null,
    "hand2_height": null,
    "hand_width": 40,
    "hold_width": 112,
    "hand_weight": 10
  },
  "auth_token": "b206db64-718f-4608-babb-0b8abe6e1b9d",
  "currency": "EUR",
  "bags": 0,
  "pending_flight_changes": [],
  "additional_bookings": {
    "details": [
      {
        "category": "bags",
        "bags": 1,
        "booked": 1491930532,
        "payment_gate": "zooz",
        "price": 23.65,
        "payment_expiration": 1491933233,
        "comment": "",
        "payment_currency": "EUR",
        "payment_token_created": 1491930533,
        "credits_payment_amount": 0,
        "payment_price": 23.65,
        "final_status": "expired",
        "data": {
          "ordered_by": "pax"
        },
        "payment_token": "K53DB52ZEIATNS4U3PJOJZALAU",
        "abid": 153678
      },
      {
        "category": "bags",
        "bags": 2,
        "booked": 1491484404,
        "payment_gate": "zooz",
        "price": 47.3,
        "payment_expiration": 1491487104,
        "comment": "",
        "payment_currency": "EUR",
        "payment_token_created": 1491484404,
        "credits_payment_amount": 0,
        "payment_price": 47.3,
        "final_status": "expired",
        "data": {
          "ordered_by": "pax"
        },
        "payment_token": "W2VQJ33VRWSAZJDIQI7MJV37XA",
        "abid": 151459
      },
      {
        "category": "bags",
        "bags": 1,
        "booked": 1491392536,
        "payment_gate": "zooz",
        "price": 23.65,
        "payment_expiration": 1491395237,
        "comment": "",
        "payment_currency": "CZK",
        "payment_token_created": 1491392537,
        "credits_payment_amount": 0,
        "payment_price": 642,
        "final_status": "expired",
        "data": {
          "ordered_by": "pax"
        },
        "payment_token": "AL73CTV7Y6AM3NPXGD4RLTL3CQ",
        "abid": 150895
      }
    ],
    "paid": {
      "bags": 0,
      "total": 0
    }
  },
  "config": {
    "allowed_to_change": {
      "bags": 36,
      "seating": null,
      "insurance_change": 48,
      "flights": 48,
      "other": 4,
      "insurance_add": 24,
      "extras": null
    },
    "allowed_to_change_segments": true
  },
  "arrival": {
    "where": {
      "code": "CDG",
      "name": "Paris"
    },
    "when": 0
  },
  "insurance_price": {
    "travel_plus": 29.14,
    "skysilver": 13.14,
    "skygold": 29.14,
    "travel_basic": 13.14
  },
  "pending_requests": [],
  "price": 69,
  "credits_payment": 0,
  "blockers": [],
  "document_options": {
    "checkin_date": 1504386600,
    "airport_checkin_price": 0,
    "document_need": 1
  },
  "passenger_ids": [
    4095455,
    4095456
  ],
  "transfers": [],
  "refund_options": [],
  "passengers": [
    {
      "category": "adults",
      "bags": 0,
      "insurance_price": null,
      "firstname": "chatbot",
      "travel_document": {
        "cardno": null,
        "expiration": null
      },
      "lastname": "test",
      "pkpass": null,
      "title": "mr",
      "hand_bags": {},
      "birthday": "1985-01-01",
      "contact_passenger": false,
      "nationality": "us",
      "id": 4095455,
      "insurance_type": "none"
    },
    {
      "category": "adults",
      "bags": 0,
      "insurance_price": null,
      "firstname": "TEST",
      "travel_document": {
        "cardno": null,
        "expiration": null
      },
      "lastname": "TEST",
      "pkpass": null,
      "title": "mr",
      "hand_bags": {},
      "birthday": "1985-01-01",
      "contact_passenger": true,
      "nationality": "us",
      "id": 4095456,
      "insurance_type": "none"
    }
  ],
  "assets": {
    "eticket": "https://skypicker-mailing.s3.amazonaws.com/2707/1490859068_E-ticket_passenger_and_1_more_e826fc3275a65a5db8808279d8fc7f8f.pdf?v=1490859069",
    "additional": [],
    "en_confirmation_link": "https://skypicker-mailing.s3.amazonaws.com/2707/1490859157_2707251_confirmation_e826fc3275a65a5db8808279d8fc7f8f.html",
    "invoice": null,
    "boardingPass": {
      "315289498": null,
      "all": null
    },
    "insurance": null,
    "transfer_ticket": null
  },
  "created": 1490858935,
  "route": [
    "LGW",
    "CDG"
  ],
  "departure": {
    "where": {
      "code": "LGW",
      "name": "London"
    },
    "when": 0
  },
  "sandbox": 1,
  "contact": {
    "phone": "+1 000",
    "pid": 4095456,
    "email": "kiwicomtester@gmail.com"
  },
  "status": "confirmed",
  "flights_history": {},
  "travel_info": {
    "4095455": {
      "0": {
        "visa": [
          {
            "status": "critical",
            "timestamp": 1447710495,
            "info": "<div class=\"content\"><p>\n</p><p>Visa required.</p>\n\n<h3>Additional information:</h3>\n<p>\n</p><p>\n</p><p>Schengen visa is also valid for French Guiana, French West Indies and Reunion, provided endorsed \"Also valid for French territories being in observation of the respective French territories\".</p>\n\n<p>\nVisitors are required to hold proof of sufficient funds to cover their stay and documents required for their next destination. \n</p>\n\n</div>",
            "country": "FR"
          }
        ],
        "health": [
          {
            "status": "ok",
            "timestamp": 1447710495,
            "info": "<div class=\"content\">This information is for guide purposes only. Other health organisations may recommend alternative precautions.<br /><br /><p>\nVaccinations not required. \n</p>\n</div>",
            "country": "FR"
          }
        ],
        "passport": [
          {
            "status": "critical",
            "timestamp": 1447710495,
            "info": "<div class=\"content\"><p>\n</p><p>Passport required.</p>\n\n<h3>Document validity rules:</h3>\n<p>\nPassports and other documents accepted for entry must be valid for a minimum of 3 months beyond the period of intended stay. \n</p>\n<h3>The following regulations apply to children/minors:</h3>\n<p>\nAlien minors up to/including 17 years of age can be included in parent's passport, provided travelling with their parents.<br />\n</p>\n<h3>Important:</h3>\n<p>\nPassports and/or passport replacing documents issued more than 10 years prior to date of travel are not accepted. \n</p>\n</div>",
            "country": "FR"
          }
        ]
      }
    },
    "4095456": {
      "0": {
        "visa": [
          {
            "status": "critical",
            "timestamp": 1447710495,
            "info": "<div class=\"content\"><p>\n</p><p>Visa required.</p>\n\n<h3>Additional information:</h3>\n<p>\n</p><p>\n</p><p>Schengen visa is also valid for French Guiana, French West Indies and Reunion, provided endorsed \"Also valid for French territories being in observation of the respective French territories\".</p>\n\n<p>\nVisitors are required to hold proof of sufficient funds to cover their stay and documents required for their next destination. \n</p>\n\n</div>",
            "country": "FR"
          }
        ],
        "health": [
          {
            "status": "ok",
            "timestamp": 1447710495,
            "info": "<div class=\"content\">This information is for guide purposes only. Other health organisations may recommend alternative precautions.<br /><br /><p>\nVaccinations not required. \n</p>\n</div>",
            "country": "FR"
          }
        ],
        "passport": [
          {
            "status": "critical",
            "timestamp": 1447710495,
            "info": "<div class=\"content\"><p>\n</p><p>Passport required.</p>\n\n<h3>Document validity rules:</h3>\n<p>\nPassports and other documents accepted for entry must be valid for a minimum of 3 months beyond the period of intended stay. \n</p>\n<h3>The following regulations apply to children/minors:</h3>\n<p>\nAlien minors up to/including 17 years of age can be included in parent's passport, provided travelling with their parents.<br />\n</p>\n<h3>Important:</h3>\n<p>\nPassports and/or passport replacing documents issued more than 10 years prior to date of travel are not accepted. \n</p>\n</div>",
            "country": "FR"
          }
        ]
      }
    }
  }
}
