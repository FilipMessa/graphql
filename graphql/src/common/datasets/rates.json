{
  "DZD": 0.0082,
  "NAD": 0.0656438,
  "GHS": 0.200187,
  "EGP": 0.0490061,
  "BGN": 0.5113,
  "PAB": 0.877821,
  "BOB": 0.126304,
  "DKK": 0.13383,
  "BWP": 0.0847094,
  "LBP": 0.000581647,
  "TZS": 0.000392217,
  "VND": 0.0000386063,
  "AOA": 0.00529968,
  "KHR": 0.000214626,
  "MYR": 0.2045,
  "SOS": 0.00149289,
  "KYD": 1.05334,
  "LYD": 0.624784,
  "UAH": 0.033762,
  "JOD": 1.23811,
  "AZN": 0.520622,
  "AWG": 0.488969,
  "SAR": 0.2341,
  "XAG": 13.4896,
  "HKD": 0.11175,
  "CHF": 0.90522,
  "GIP": 1.12935,
  "BYR": 0.0000458628,
  "SVC": 0.100324,
  "XPD": 733.829,
  "BYN": 0.44413,
  "MRO": 0.00242325,
  "HRK": 0.1351,
  "DJF": 0.00491034,
  "THB": 0.0257,
  "XAF": 0.00152356,
  "BND": 0.644,
  "ETB": 0.0375177,
  "UYU": 0.0303065,
  "NIO": 0.0292509,
  "LAK": 0.000106403,
  "SYP": 0.00409559,
  "MAD": 0.090879,
  "MZN": 0.0146295,
  "YER": 0.00350673,
  "ZAR": 0.0655,
  "NPR": 0.0086,
  "ZWL": 0.00272288,
  "NGN": 0.00250806,
  "CRC": 0.00154515,
  "AED": 0.2427,
  "GBP": 1.12572,
  "MWK": 0.00120964,
  "LKR": 0.0058,
  "PKR": 0.00831073,
  "HUF": 0.0032,
  "SZL": 0.0656806,
  "LSL": 0.0656948,
  "MNT": 0.000367879,
  "AMD": 0.00182673,
  "UGX": 0.000243832,
  "XDR": 1.21782,
  "JMD": 0.00678989,
  "TTD": 0.1326,
  "SHP": 1.12935,
  "AFN": 0.0128468,
  "MMK": 0.000645932,
  "KPW": 0.000975259,
  "TRY": 0.2432,
  "BDT": 0.0107748,
  "HTG": 0.014054,
  "SLL": 0.00011767,
  "MGA": 0.000292851,
  "PHP": 0.0173,
  "LRD": 0.00933889,
  "XCD": 0.324812,
  "NOK": 0.10483,
  "MOP": 0.10907,
  "INR": 0.0136,
  "MXN": 0.0489,
  "CZK": 0.03815,
  "TJS": 0.0996386,
  "BTC": 2141.0,
  "BTN": 0.0135872,
  "COP": 0.000284868,
  "TMT": 0.250093,
  "MUR": 0.0256,
  "IDR": 0.0000654754,
  "GEL": 0.366804,
  "XPF": 0.00836387,
  "FJD": 0.42854,
  "VUV": 0.0082582,
  "PEN": 0.272,
  "BZD": 0.436723,
  "ILS": 0.247,
  "DOP": 0.0184804,
  "GGP": 1.12935,
  "MDL": 0.048394,
  "XPT": 786.508,
  "QAR": 0.2449,
  "SEK": 0.10353,
  "ZMK": 0.000174842,
  "MVR": 0.0565969,
  "AUD": 0.66392,
  "SRD": 0.117238,
  "CUP": 0.0344244,
  "CLF": 35.4933,
  "BBD": 0.438911,
  "KMF": 0.00203576,
  "KRW": 0.000762812,
  "GMD": 0.0190831,
  "VEF": 0.0879138,
  "GTQ": 0.119764,
  "ANG": 0.493156,
  "CUC": 0.877821,
  "CLP": 0.0013,
  "ZMW": 0.0972116,
  "EUR": 1.0,
  "ALL": 0.0075492,
  "RWF": 0.00106338,
  "KZT": 0.00268533,
  "RUB": 0.0140456,
  "ISK": 0.00838243,
  "CDF": 0.000574302,
  "RON": 0.2188,
  "OMR": 2.318,
  "BRL": 0.2682,
  "SBD": 0.111728,
  "PLN": 0.23508,
  "PYG": 0.000158377,
  "KES": 0.0085,
  "QUN": 0.135584,
  "MKD": 0.0162357,
  "EEK": 0.0639,
  "TWD": 0.0287048,
  "TOP": 0.38996,
  "JEP": 1.12935,
  "GNF": 0.0000975357,
  "WST": 0.347559,
  "IQD": 0.000742343,
  "ERN": 0.0569018,
  "BAM": 0.511223,
  "SCR": 0.0647433,
  "CAD": 0.67751,
  "CVE": 0.0090684,
  "KWD": 2.9377,
  "BIF": 0.000508808,
  "PGK": 0.27654,
  "LTL": 0.2896,
  "HNL": 0.0374196,
  "SGD": 0.6337,
  "UZS": 0.000220005,
  "STD": 0.0000408272,
  "IRR": 0.0000269556,
  "CNY": 0.1291,
  "XOF": 0.00152126,
  "TND": 0.3642,
  "GYD": 0.00421905,
  "MTL": 1.36902,
  "NZD": 0.6381,
  "FKP": 1.12935,
  "LVL": 1.4229,
  "USD": 0.87313,
  "BSD": 0.877821,
  "KGS": 0.0126106,
  "ARS": 0.0517885,
  "BMD": 0.877821,
  "IMP": 1.12935,
  "RSD": 0.0083,
  "BHD": 2.3708,
  "JPY": 0.00764,
  "SDG": 0.13141,
  "XAU": 1061.88
}