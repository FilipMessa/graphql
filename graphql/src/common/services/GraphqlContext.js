// @flow

import DataLoader from 'dataloader';

import IdentityDataloader from '../../identity/dataloaders/Identity';
import createBookingLoader from '../../booking/dataloaders/Booking';
import createAirlineLoader from '../../flight/dataloaders/Airline';
import createRatesLoader from '../dataloaders/Rates';
import BookingsLoader from '../../booking/dataloaders/Bookings';
import LocationSuggestionsLoader from '../../location/dataloaders/LocationSuggestions';
import LocationLoader from '../../location/dataloaders/Location';
import FlightLoader from '../../flight/dataloaders/Flight';
import OptionsStorage from './context/OptionsStorage';
import HotelsAvailability from '../../hotel/dataloaders/HotelsAvailability';
import { type SearchParameters as HotelKey } from '../../hotel/dataloaders/flow/SearchParameters';
import HotelByID from '../../hotel/dataloaders/HotelByID';
import HotelCities from '../../hotel/dataloaders/HotelCities';
import HotelRoomsLoader from '../../hotel/dataloaders/HotelRooms';
import HotelRoomAvailabilityLoader from '../../hotel/dataloaders/HotelRoomAvailability';
import HotelRoomBeddingLoader from '../../hotel/dataloaders/RoomBedding';
import PriceStatsLoader from '../../hotel/dataloaders/PriceStats';

import type { Booking } from '../../booking/Booking';
import type { Airline } from '../../flight/Flight';
import type { HotelType } from '../../hotel/dataloaders/flow/HotelType';
import type { HotelCity } from '../../hotel/types/outputs/HotelCity';

/**
 * FIXME: this is stupid - it's already defined by data-loader itself
 */
export type GraphqlContextType = {|
  // DataLoader<K, V>
  apiToken: ?string,
  dataLoader: {|
    airline: DataLoader<string, ?Airline>,
    booking: DataLoader<number | string, Booking>,
    bookings: BookingsLoader,
    flight: FlightLoader,
    identity: IdentityDataloader,
    location: LocationLoader,
    locationSuggestions: LocationSuggestionsLoader,
    rates: DataLoader<string, ?number>,
    hotel: {
      availabilityByLocation: DataLoader<HotelKey, HotelType[]>,
      availabilityByID: DataLoader<HotelKey, HotelType[]>,
      byID: typeof HotelByID,
      cities: DataLoader<string, HotelCity[]>,
      room: HotelRoomsLoader,
      roomAvailability: HotelRoomAvailabilityLoader,
      roomBedding: typeof HotelRoomBeddingLoader,
      priceStats: DataLoader<
        { searchParams: HotelKey, boundary: 'MAX' | 'MIN' },
        number,
      >,
    },
  |},
  options: OptionsStorage,
  _traceCollector?: Object,
|};

export function createContext(token: ?string): GraphqlContextType {
  const bookings = new BookingsLoader(token);
  const locationSuggestions = new LocationSuggestionsLoader();
  const location = new LocationLoader(locationSuggestions);

  return {
    apiToken: token,
    dataLoader: {
      airline: createAirlineLoader(),
      booking: createBookingLoader(token, bookings),
      bookings: bookings,
      flight: new FlightLoader(location),
      identity: new IdentityDataloader(token),
      location: location,
      locationSuggestions: locationSuggestions,
      rates: createRatesLoader(),
      hotel: {
        availabilityByLocation: HotelsAvailability,
        availabilityByID: HotelsAvailability,
        byID: HotelByID,
        cities: HotelCities,
        room: new HotelRoomsLoader(),
        roomAvailability: new HotelRoomAvailabilityLoader(),
        roomBedding: HotelRoomBeddingLoader,
        priceStats: PriceStatsLoader,
      },
    },
    options: new OptionsStorage(),
  };
}
