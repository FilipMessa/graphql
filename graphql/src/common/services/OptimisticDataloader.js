// @flow

import DataLoader from 'dataloader';

// any because it must be very generic (accepts everything)
type K = any;
type V = any;

/**
 * Original data-loader kills load batch if one of this response values is
 * Error instance. This optimistic data-loader keeps Error value as a valid
 * response. This means that it's possible to return as many responses
 * as possible and convert Errors to the error messages in the response
 * if necessary.
 *
 * @see https://github.com/facebook/dataloader/issues/41
 */
export default class OptimisticDataloader extends DataLoader<K, V> {
  constructor(batchLoadFn: Function) {
    super(batchLoadFn);
  }

  loadMany = (keys: K[]): Promise<V[]> => {
    if (!Array.isArray(keys)) {
      throw new TypeError(
        'The loader.loadMany() function must be called with Array<key> ' +
          `but got: ${keys}.`,
      );
    }

    return Promise.all(
      keys.map(key =>
        this.load(key).catch(error => {
          if (error instanceof TypeError) {
            // original data-loader throws TypeErrors
            throw error;
          }
          return error;
        }),
      ),
    );
  };
}
