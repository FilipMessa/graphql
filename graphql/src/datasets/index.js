// @flow

module.exports.Booking = {
  all: require('./booking/AllBookings.json'),
  '2707224': require('./booking/booking-2707224.json'),
  '2707229': require('./booking/booking-2707229.json'),
  '2707251': require('./booking/booking-2707251.json'),
};

module.exports.Location = {
  unknown: require('./location/unknown.json'),
  prague: require('./location/prague.json'),
  pragueCsCZ: require('./location/prague-cs-CZ.json'),
  mexico: require('./location/mexico.json'),
  mexCsCZ: require('./location/mex-cs-CZ.json'),
  frankfurt: require('./location/frankfurt.json'),
};

module.exports.Flight = {
  prgMex: require('./flight/prg-mex.json'),
  prgMexCzk: require('./flight/prg-mex-czk.json'),
  prgFraMex: require('./flight/prg,fra-mex.json'),
  prgMexCsCZ: require('./flight/prg-mex-cs-CZ.json'),
  noResults: require('./flight/no-results.json'),
  prgMexFrom7To10Days: require('./flight/prg-mex-from-7-to-10-days.json'),
};

module.exports.Airline = {
  all: require('./airline/airlines.json'),
};

module.exports.Identity = {
  currentUser: require('./identity/currentUser.json'),
};
