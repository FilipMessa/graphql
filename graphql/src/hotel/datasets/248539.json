[
  {
    "pagename": "sheraton-milan-malpensa-airport-conference-centre",
    "currencycode": "EUR",
    "created": "2010-06-08 09:41:07",
    "book_domestic_without_cc_details": "0",
    "contractchain_id": "0",
    "creditcard_required": "1",
    "city_id": "900048081",
    "zip": "21010",
    "class": "4",
    "class_is_estimated": "0",
    "hoteltype_id": "14",
    "preferred": "1",
    "name": "Sheraton Milan Malpensa Airport Hotel & Conference Centre",
    "city": "Case Nuove",
    "exact_class": "4.0",
    "languagecode": "en",
    "maxrate": "787",
    "district": "",
    "countrycode": "it",
    "is_closed": "0",
    "nr_rooms": "437",
    "address": "Malpensa Terminal 1 S.S. 336",
    "minrate": "105",
    "url": "https://www.booking.com/hotel/it/sheraton-milan-malpensa-airport-conference-centre.html",
    "max_persons_in_reservation": "0",
    "hotel_id": "248539",
    "commission": "0",
    "max_rooms_in_reservation": "8",
    "facilities": [
      {
        "name": "Restaurant",
        "facilitytype_id": "7",
        "languagecode": "en",
        "hotelfacilitytype_id": "3",
        "type": "boolean",
        "hotel_id": "248539"
      },
      {
        "type": "boolean",
        "hotelfacilitytype_id": "5",
        "facilitytype_id": "7",
        "name": "Room service",
        "languagecode": "en",
        "hotel_id": "248539"
      },
      {
        "hotelfacilitytype_id": "6",
        "languagecode": "en",
        "facilitytype_id": "27",
        "name": "Meeting/banquet facilities",
        "type": "free_or_paid",
        "hotel_id": "248539"
      }
    ],
    "rooms": [
      {
        "roomtype": "Twin/Double",
        "max_persons": "3",
        "hotel_id": "248539",
        "bookable": "1",
        "min_price": "96.3",
        "roomtype_id": "24",
        "ranking": "1000",
        "max_price": "796.14",
        "room_id": "24853901",
        "bedding": [
          {
            "type": "Single bed(s)",
            "amount": "2"
          },
          {
            "type": "Extra-large double bed(s) (Super-king size)",
            "amount": "1"
          }
        ]
      },
      {
        "min_price": "0",
        "bookable": "1",
        "roomtype": "Twin/Double",
        "max_persons": "3",
        "hotel_id": "248539",
        "max_price": "0",
        "room_id": "24853903",
        "roomtype_id": "24",
        "ranking": "1002",
        "bedding": [
          {
            "amount": "2",
            "type": "Single bed(s)"
          },
          {
            "type": "Extra-large double bed(s) (Super-king size)",
            "amount": "1"
          }
        ]
      }
    ],
    "descriptions": [
      {
        "description": "Sheraton Milan Malpensa Hotel is located inside Malpensa Airport's Terminal 1. It offers air-conditioned rooms with minibar and flat-screen TV. A fitness centre is also available.\n\nRooms at Sheraton Milano Malpensa Airport Hotel and Conference Centre are soundproofed and feature Sheraton Sweet Sleeper beds.\n\nOther hotel facilities include business meeting rooms, a spa and beauty centre and indoor swimming pool.\n\nGuests can sip a cocktail at the on-site bar and have dinner at the restaurant, which specialises in Italian cuisine.\n\nThe hotel is a 40-minute drive from Milan centre, and 40 km from Rho Fiera Milano Exhibition Centre. You can also use public and private shuttles to Milan Central Railway Station.",
        "languagecode": "en",
        "hotel_id": "248539",
        "descriptiontype_id": "6"
      },
      {
        "description": "Please note that the spa is closed from 27 April until 14 May. Please inform Sheraton Milan Malpensa Airport Hotel & Conference Centre in advance of your expected arrival time. You can use the Special Requests box when booking, or contact the property directly with the contact details provided in your confirmation.",
        "hotel_id": "248539",
        "languagecode": "en",
        "descriptiontype_id": "7"
      }
    ],
    "photos": [
      {
        "url_max300": "http://aff.bstatic.com/images/hotel/max300/267/26753013.jpg",
        "hotel_id": "248539",
        "url_original": "http://aff.bstatic.com/images/hotel/max500/267/26753013.jpg",
        "url_square60": "http://aff.bstatic.com/images/hotel/square60/267/26753013.jpg",
        "photo_id": "26753013"
      }
    ],
    "chains": [
      {
        "chain_id": "1056",
        "name": "Sheraton",
        "hotel_id": "248539"
      }
    ],
    "checkin": {
      "to": "",
      "from": "15:00"
    },
    "checkout": {
      "to": "",
      "from": "15:00"
    },
    "location": {
      "latitude": "45.6275442354691",
      "longitude": "8.71018409729004"
    }
  }
]
