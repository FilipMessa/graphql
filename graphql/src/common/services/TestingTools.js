// @flow

import { graphql as graphqlNodeLibrary } from 'graphql';
import schema from '../../Schema';
import { createContext } from './GraphqlContext';

/**
 * TestingTools must be imported FIRST otherwise mock of HttpRequest won't work.
 */
jest.mock('../services/HttpRequest');
const mockedHttpRequest = require('./HttpRequest');

class MockResponse {
  absoluteUrl: string;
  httpMethod: string;

  constructor(httpMethod: 'GET' | 'POST', absoluteUrl: string) {
    this.httpMethod = httpMethod;
    this.absoluteUrl = absoluteUrl;
  }

  replyWithData(response: Array<Object> | Object) {
    // $FlowAllowMockMagic
    mockedHttpRequest.__setMockData(
      this.httpMethod,
      this.absoluteUrl,
      response,
    );
  }
}

export class RestApiMock {
  static onGet(absoluteUrl: string) {
    return new MockResponse('GET', absoluteUrl);
  }

  static onPost(absoluteUrl: string) {
    return new MockResponse('POST', absoluteUrl);
  }
}

export const graphql = async (
  query: string,
  variables: ?Object,
): Promise<Object> =>
  graphqlNodeLibrary(
    schema,
    query,
    null,
    createContext('test_token'),
    variables,
  );

export const evaluateResolver = (field: mixed, testValue: mixed) => {
  // $FlowAllowNextLineInThisTest (fields are possibly undefined but we assume it's fine)
  return field.resolve(testValue);
};
