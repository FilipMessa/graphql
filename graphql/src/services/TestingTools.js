import { graphql } from 'graphql';
import schema from '../Schema';
import { createContext } from '../services/GraphqlContext';

jest.mock('../services/HttpRequest');

export const executeQuery = async (query: string): Promise<Object> =>
  graphql(schema, query, null, createContext('test_token'));
