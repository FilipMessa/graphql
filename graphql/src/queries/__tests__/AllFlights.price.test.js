// @flow

import { graphql, RestApiMock } from '../../services/TestingTools';
import config from '../../../config/application';
import { Flight } from '../../datasets';

RestApiMock.onGet(
  config.restApiEndpoint.allFlights({
    flyFrom: 'PRG',
    to: 'MEX',
    dateFrom: '08/08/2017',
    dateTo: '08/09/2017',
  }),
).replyWithData(Flight.prgMex);

RestApiMock.onGet(
  config.restApiEndpoint.allFlights({
    flyFrom: 'PRG',
    to: 'MEX',
    dateFrom: '08/08/2017',
    dateTo: '08/09/2017',
    curr: 'CZK',
  }),
).replyWithData(Flight.prgMexCzk);

describe('all flights query', () => {
  it('should return flight prices', async () => {
    const allFlightsSearchQuery = `
    query ($input: FlightsSearchInput!) {
      allFlights(search: $input) {
        edges {
          node {
            price { amount, currency }
          }
        }
      }
    }`;
    const variables = {
      input: {
        from: {
          location: 'PRG',
        },
        to: {
          location: 'MEX',
        },
        dateFrom: {
          exact: '2017-08-08',
        },
        dateTo: {
          exact: '2017-09-08',
        },
      },
    };
    expect(await graphql(allFlightsSearchQuery, variables)).toMatchSnapshot();
  });

  it('should return flight prices in different currency', async () => {
    const allFlightsSearchQuery = `
    query ($input: FlightsSearchInput!, $options: FlightsOptionsInput) {
      allFlights(search: $input, options: $options) {
        edges {
          node {
            price { amount, currency }
          }
        }
      }
    }`;
    const variables = {
      input: {
        from: {
          location: 'PRG',
        },
        to: {
          location: 'MEX',
        },
        dateFrom: {
          exact: '2017-08-08',
        },
        dateTo: {
          exact: '2017-09-08',
        },
      },
      options: {
        currency: 'CZK',
      },
    };
    expect(await graphql(allFlightsSearchQuery, variables)).toMatchSnapshot();
  });
});
