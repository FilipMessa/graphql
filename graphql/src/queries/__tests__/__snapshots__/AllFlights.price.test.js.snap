// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all flights query should return flight prices 1`] = `
Object {
  "data": Object {
    "allFlights": Object {
      "edges": Array [
        Object {
          "node": Object {
            "price": Object {
              "amount": 277,
              "currency": "EUR",
            },
          },
        },
        Object {
          "node": Object {
            "price": Object {
              "amount": 294,
              "currency": "EUR",
            },
          },
        },
        Object {
          "node": Object {
            "price": Object {
              "amount": 300,
              "currency": "EUR",
            },
          },
        },
      ],
    },
  },
}
`;

exports[`all flights query should return flight prices in different currency 1`] = `
Object {
  "data": Object {
    "allFlights": Object {
      "edges": Array [
        Object {
          "node": Object {
            "price": Object {
              "amount": 8606,
              "currency": "CZK",
            },
          },
        },
        Object {
          "node": Object {
            "price": Object {
              "amount": 8616,
              "currency": "CZK",
            },
          },
        },
        Object {
          "node": Object {
            "price": Object {
              "amount": 8622,
              "currency": "CZK",
            },
          },
        },
      ],
    },
  },
}
`;
